### PR TITLE
Added Reverse Proxy for Anthropic

### PR DESCRIPTION
**Pull request by madonchik123**
## Summary

Made it so you can specify reverse proxy for anthropic

## Change Type

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation update 

## Testing

I've took  this reverse proxy for testing https://discordexperiment.replit.app
and my own api key which you can get by sending an request to it without using any api key but for testing purposes just leave 1 as a key
change ANTHROPIC_REVERSE_PROXY= to ANTHROPIC_REVERSE_PROXY=https://discordexperiment.replit.app
and done

## Checklist

- [x]  My code adheres to this project's style guidelines
- [x]  I have performed a self-review of my own code
- [x]  I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x]  My changes do not introduce new warnings
- [x]  I have written tests demonstrating that my changes are effective or that my feature works
- [x]  Local unit tests pass with my changes
- [x]  Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -97,7 +97,7 @@ DEBUG_OPENAI=false # Set to true to enable debug mode for the OpenAI endpoint
 # Identify the available models, separated by commas *without spaces*. 
 # The first will be default. 
 # Leave it blank to use internal settings. 
-# OPENAI_MODELS=gpt-3.5-turbo,gpt-3.5-turbo-16k,gpt-3.5-turbo-0301,text-davinci-003,gpt-4,gpt-4-0314,gpt-4-0613
+OPENAI_MODELS=gpt-3.5-turbo,gpt-3.5-turbo-16k,gpt-3.5-turbo-0301,text-davinci-003,gpt-4,gpt-4-0314,gpt-4-0613
```
Comment (danny-avila) [inline .env.example]: this needs to stay uncommented

File: api/app/clients/AnthropicClient.js [hunk 2]
```diff
@@ -9,10 +9,13 @@ const AI_PROMPT = '\n\nAssistant:';
 const tokenizersCache = {};
 
 class AnthropicClient extends BaseClient {
-  constructor(apiKey, options = {}, cacheOptions = {}) {
+  constructor(apiKey, options = {}, cacheOptions = {},baseURL) {
```
Comment (danny-avila) [inline api/app/clients/AnthropicClient.js]: there are some linting errors here and a few other places

File: api/server/routes/endpoints/anthropic/initializeClient.js [hunk 3]
```diff
@@ -16,7 +15,12 @@ const initializeClient = async ({ req, res }) => {
     key = await getUserKey({ userId: req.user.id, name: 'anthropic' });
   }
   let anthropicApiKey = isUserProvided ? key : ANTHROPIC_API_KEY;
-  const client = new AnthropicClient(anthropicApiKey, { req, res });
+  if (typeof ANTHROPIC_REVERSE_PROXY === 'undefined') {
+    Reverse_Proxy = undefined;
```
Comment (danny-avila) [inline api/server/routes/endpoints/anthropic/initializeClient.js]: you should not be using a global variable

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: api/server/routes/endpoints/anthropic/initializeClient.js (hunk 3 above)
Comment (madonchik123) [reply api/server/routes/endpoints/anthropic/initializeClient.js:19]: my bad one sec lemme use let instead gonna fix comments aswell

**Comment by danny-avila** [comment]
Hi @madonchik123 Thanks for your PR

I don't see any changes/resolutions since you requested review again. Let me know if you need any clarity here

File: api/server/routes/endpoints/anthropic/initializeClient.js [hunk 4]
```diff
@@ -2,9 +2,8 @@ const { AnthropicClient } = require('../../../../app');
 const { getUserKey, checkUserKeyExpiry } = require('../../../services/UserService');
 
 const initializeClient = async ({ req, res }) => {
-  const { ANTHROPIC_API_KEY } = process.env;
+  const { ANTHROPIC_API_KEY,ANTHROPIC_REVERSE_PROXY } = process.env;
```
Comment (danny-avila) [inline api/server/routes/endpoints/anthropic/initializeClient.js]: make sure you have eslint installed. Please read the contributing guide for best practices here: https://github.com/danny-avila/LibreChat/blob/main/.github/CONTRIBUTING.md

**Review by danny-avila — CHANGES_REQUESTED** [review]

**Comment by madonchik123** [comment]
> Hi @madonchik123 Thanks for your PR
> 
> I don't see any changes/resolutions since you requested review again. Let me know if you need any clarity here

Oof I quite didn't understand what I needed to change,because i've never made a PR before. I would really appreciate if you could like tell what exactly to change i've tried to use lint but it says failed and other things but its not because of my code though

**Comment by Berry-13** [comment]
@madonchik123 please test if this still works

**Comment by madonchik123** [comment]
Yes seems to work

**Review by danny-avila — APPROVED** [review]

**Review by danny-avila — APPROVED** [review]

**Pull request merged**